server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1047 @@
+        if content_type is None:
+            raise MissingValue('content_type')
+        link = add_link_profile(new_profile)
+        response = generate_json_response_with_pulp_encoder(new_profile)
+        redirect_response = generate_redirect_response(response, link['_href'])
+        return redirect_response
+
+
+class ConsumerProfileResourceView(View):
+    """
+    View Consumer profiles represents the collection of
+    objects used to associate consumer and installed content
+    unit profiles.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, consumer_id, content_type):
+        """
+        Get profile by content type associated with consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: The consumer ID.
+        :type consumer_id: str
+        :param content_type: The content type
+        :type consumer_id: str
+
+        :return: Response representing consumer's profile
+        :rtype: django.http.HttpResponse
+        """
+        manager = factory.consumer_profile_manager()
+        profile = manager.get_profile(consumer_id, content_type)
+        add_link_profile(profile)
+        return generate_json_response_with_pulp_encoder(profile)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_required
+    def put(self, request, consumer_id, content_type):
+        """
+        Update the association of a profile with a consumer by content type ID.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+        :param content_type: A content unit type ID.
+        :type content_type: str
+
+        :return: Response representing the updated profile
+        :rtype: django.http.HttpResponse
+        """
+        body = request.body_as_json
+        profile = body.get('profile')
+
+        manager = factory.consumer_profile_manager()
+        consumer = manager.update(consumer_id, content_type, profile)
+
+        add_link_profile(consumer)
+
+        return generate_json_response_with_pulp_encoder(consumer)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, consumer_id, content_type):
+        """
+        Delete an association between the specified
+        consumer and profile.  Designed to be idempotent.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+        :param content_type: The content type ID.
+        :type content_type: str
+
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        """
+        manager = factory.consumer_profile_manager()
+        return generate_json_response(manager.delete(consumer_id, content_type))
```

Comment:
I'm not really a fan of this pattern, but it is a style thing so feel free to leave as is. 

What I dislike is that it is not obvious that manager.delete() does not return anything. You could make the argument that if we ever wanted to change the manager to return something, you could ignore this file, but I think that scenario makes it more likely that the doc block becomes incorrect over time.